debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -110,7 +131,25 @@ public static ConnectionFactory patternBasedFactory(String urlPattern, Field...
             String url = findAndReplace(urlPattern, props, varsWithDefaults);
             LOGGER.trace("Props: {}", props);
             LOGGER.trace("URL: {}", url);
-            Connection conn = DriverManager.getConnection(url, props);
+            
+            Connection conn = null;
+            String driverClassName = config.getString(JdbcConfiguration.DRIVER);
+            if (driverClassName == null) {
+                conn = DriverManager.getConnection(url, props);
+            } else {
+                try {
+                    Class<java.sql.Driver> driverClazz = null;
+                    if (classloader == null) {
+                        driverClazz = (Class<java.sql.Driver>) Class.forName(driverClassName);
+                    } else {
+                        driverClazz = (Class<java.sql.Driver>) Class.forName(driverClassName, true, classloader);
+                    }
```

Comment:
This probably isn't as consistent with `patternBasedFactory(String, Field...)`. Seems like that method could pass null into the `classloader` parameter, and then this method should set `classloader = JdbcConnection.class.getClassLoader()` when `classloader` is null.